src/components/common/Button/Button.tsx

```diff
@@ -1,23 +1,15 @@
 import React from 'react';
 import * as styled from './Button.styles';
 
-type ButtonProps = Omit<
-  React.DetailedHTMLProps<React.ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>,
-  'ref'
-> &
-  React.PropsWithChildren<{
-    variant?: 'primary' | 'outlined';
-  }>;
+type ButtonProps = React.ButtonHTMLAttributes<HTMLButtonElement> &
+  React.PropsWithChildren<{ variant?: 'primary' | 'outlined' }>;
 
-class Button extends React.PureComponent<ButtonProps> {
-  render() {
-    const { children, variant, ...buttonProps } = this.props;
-    return (
-      <styled.Button {...buttonProps} $variant={variant ?? 'outlined'}>
-        {children}
-      </styled.Button>
-    );
-  }
-}
+const Button = ({ children, variant, ...buttonProps }: ButtonProps) => {
+  return (
+    <styled.Button {...buttonProps} $variant={variant ?? 'outlined'}>
```

Comment:
```suggestion
const Button = ({ children, variant = 'outlined', ...buttonProps }: ButtonProps) => {
  return (
    <styled.Button {...buttonProps} $variant={variant}>
```

default parameter를 써서 요렇게 쓰는건 어떨까요?

어떤게 더 낫다고 생각하세요? (궁금)

Reply:
고민을 해봤는데, 저는 `default param` 을 부여하는 후자의 경우가 더 깔끔하지 않은가..? 생각이 들었습니다! 아래와 리팩토링 진행했습니다! 

```ts
const Button = ({ children, variant = 'outlined', ...buttonProps }: ButtonProps) => {
  return (
    <styled.Button {...buttonProps} $variant={variant}>
      {children}
    </styled.Button>
  );
};
```